packages/transformers/src/tests/idiomatic-test.ts

```diff
@@ -0,0 +1,657 @@
+import {compileTsFragment, CompilerHostCache} from './compile-ts-fragment.js';
+import * as ts from 'typescript';
+import * as assert from 'uvu/assert';
+import * as prettier from 'prettier';
+import idiomaticLitDecoratorTransformer from '../idiomatic.js';
+
+const cache = new CompilerHostCache();
+
+/**
+ * Compile the given fragment of TypeScript source code using the idiomatic
+ * decorator transform. Check that there are no errors and that the output
+ * matches (prettier-formatted).
+ */
+function checkTransform(inputTs: string, expectedJs: string) {
+  const options = ts.getDefaultCompilerOptions();
+  options.target = ts.ScriptTarget.ESNext;
+  options.module = ts.ModuleKind.ESNext;
+  options.moduleResolution = ts.ModuleResolutionKind.NodeJs;
+  options.importHelpers = true;
+  // Don't automatically load typings from nodes_modules/@types, we're not using
+  // them here, so it's a waste of time.
+  options.typeRoots = [];
+  options.experimentalDecorators = true;
+  const result = compileTsFragment(
+    inputTs,
+    __dirname,
+    options,
+    cache,
+    (program) => ({
+      before: [idiomaticLitDecoratorTransformer(program)],
+    })
+  );
+
+  let formattedExpected = prettier.format(expectedJs, {parser: 'typescript'});
+  // TypeScript >= 4 will add an empty export statement if there are no imports
+  // or exports to ensure this is a module. We don't care about checking this.
+  const unformattedActual = (result.code || '').replace('export {};', '');
+  let formattedActual;
+  try {
+    formattedActual = prettier.format(unformattedActual, {
+      parser: 'typescript',
+    });
+  } catch {
+    // We might emit invalid TypeScript in a failing test. Rather than fail with
+    // a Prettier parse exception, it's more useful to see a diff.
+    formattedExpected = expectedJs;
+    formattedActual = unformattedActual;
+  }
+  assert.is(formattedActual, formattedExpected);
+  assert.equal(
+    result.diagnostics.map((diagnostic) =>
+      ts.formatDiagnostic(diagnostic, result.host)
+    ),
+    []
+  );
+}
+
+test('@customElement', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {customElement} from 'lit/decorators.js';
+  @customElement('my-element')
+  class MyElement extends LitElement {
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+  }
+  customElements.define('my-element', MyElement);
+  `;
+  checkTransform(input, expected);
+});
+
+test('@property', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {property} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @property()
+    str = "foo";
+
+    @property({type: Number, attribute: false})
+    num = 42;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    constructor() {
+      super(...arguments);
+      this.str = "foo";
+      this.num = 42;
+    }
+    static get properties() {
```

Comment:
Can we have an option to email `static properties = {...}`?

Reply:
Discussed offline. Decided to leave as the getter for now, since that's what we document for JS at https://lit.dev/. Can revisit later.